tests/main/snap-remove-terminate/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Ensure that the remove --terminate flag kills running snap apps.
+
+details: |
+    This test spawns a snap app background process and checks that it
+    gets terminated when removing with the --terminate flag.
+
+systems:
+    # Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+    - -ubuntu-14.04-*
+
+prepare: |
+    snap install test-snapd-sh
+
+    # TODO: Discard api-client when --terminate flag is added to remove command
+    "$TESTSTOOLS"/snaps-state install-local api-client
+    snap connect api-client:snapd-control
+
+restore: |
+    snap remove --purge api-client
+
+execute: |
+    echo "Start a long running process"
+    test-snapd-sh.sh -c "sleep 100000" &
+    pid=$!
+    # Make sure process is alive
+    kill -s 0 "$pid"
+    # Clean running process if test failed
+    tests.cleanup defer "kill $pid 2>/dev/null || true"
+
+    echo "Remove snap with --terminate flag"
+    # TODO: replace with "snap remove --terminate test-snapd-sh" when cli flag is added
+    sudo api-client --method=POST /v2/snaps/test-snapd-sh '{"action":"remove","terminate":true}'
+
+    echo "Running process should be terminated after remove change is complete"
+    retry -n 10 not kill -s 0 "$pid"
```

Comment:
Somewhat of a nitpick but this is just as racy. Perhaps the process should be a flock and we should ensure that the locked file is no longer locked. Together with systemd-run we could make this test PID-free.

Reply:
Thanks for the suggestion, the test is much cleaner/stable now